jazz_deployments/components/utils.js

```diff
@@ -0,0 +1,243 @@
+
+var initDynamodb = function () {
+    AWS.config.update({
+        region: global.config.ddb_region
+    });
+    var dynamodb = new AWS.DynamoDB();
+
+    return dynamodb;
+};
+
+var isEmpty = function (obj) {
+    for (var prop in obj) {
+        if (obj.hasOwnProperty(prop)) return false;
+    }
+    return true;
+};
+
+// convert object returned from the database, as per schema
+var formatData = function (data, format) {
+    if (!data) {
+        return {};
+    }
+
+    var deployment_obj = {};
+
+    var parseValue = function (value) {
+        var type = Object.keys(value)[0];
+        var parsed_value = value[type];
+        if (type === "NULL") {
+            return null;
+        } else if (type === "N") {
+            return Number(value);
+        } else if (type === "NS") {
+            return parsed_value.map(Number);
+        } else if (type === "S") {
+            return parsed_value;
+        } else if (type === "SS") {
+            return parsed_value;
+        } else if (type === "L") {
+            var parsed_value_list = [];
+            try {
+                for (var i = 0; i < parsed_value.length; i++) {
+                    parsed_value_list.push(parseValue(parsed_value[i]));
+                }
+            } catch (e) {}
+            return parsed_value_list;
+        } else {
+            // probably should be error
+            return parsed_value;
+        }
+    };
+
+    Object.keys(data).map(function (key) {
+        var key_name = getSchemaKeyName(key);
+        var value = data[key];
+        if (value) {
+            if (format) {
+                deployment_obj[key_name] = parseValue(value);
+            } else {
+                deployment_obj[key_name] = value;
+            }
+        }
+    });
+
+    return deployment_obj;
+};
+
+var getSchemaKeyName = function (key) {
+    // Convert database key name back, as per schema
+
+    if (!key) {
+        return null;
+    }
+
+    if (key === "SERVICE_NAME") {
+        return "service";
+    } else if (key === "DEPLOYMENT_STATUS") {
+        return "status";
+    } else if (key === "DOMAIN_NAME") {
+        return "domain";
+    } else {
+        return key.toLowerCase();
+    }
+};
+
+var ConvertKeysToLowerCase = function (obj) {
+    var output = {};
+    for (i in obj) {
+        if (Object.prototype.toString.apply(obj[i]) === '[object Object]') {
+            output[i.toLowerCase()] = ConvertKeysToLowerCase(obj[i]);
+        } else if (Object.prototype.toString.apply(obj[i]) === '[object Array]') {
+            output[i.toLowerCase()] = [];
+            output[i.toLowerCase()].push(ConvertKeysToLowerCase(obj[i][0]));
+        } else {
+            output[i.toLowerCase()] = obj[i];
+        }
+    }
+    return output;
+};
+
+// function to convert key name in schema to database column name
+var getDeploymentDatabaseKeyName = function (key) {
+    // Some of the keys in schema may be reserved keywords, so it may need some manipulation
+
+    if (!key) {
```

Comment:
this code is being repeated from lines 110. please look at previous comment on how to make it better

Reply:
Done